### PR TITLE
[GPU] Update strided slice impl for non-constant begin/end/stride support

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - Update primitive API for non-constant begin/end/stride support
 - Update impl and strided slice ref kernel w/o kernel params from memory_deps
 - Remove existing W/A for strided slice with non-constant begin/end/stride
 - Add strided slice dynamic TCs for gpuFunctests

### Tickets:
 - 94606

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/kernels/strided_slice/strided_slice_kernel_ref.h [hunk 1]
```diff
@@ -8,6 +8,13 @@
 #include <vector>
 
 namespace kernel_selector {
+
+enum class StridedSliceArgType {
+    None,
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/strided_slice/strided_slice_kernel_ref.h:13]: remove

File: src/plugins/intel_gpu/src/graph/impls/ocl/strided_slice.cpp [hunk 2]
```diff
@@ -103,34 +136,36 @@ struct strided_slice_impl : typed_primitive_impl_ocl<strided_slice> {
...
-            // E.g. out_shape[dim] = 100; begin=0; end=100; stride=-1
-            // swap: begin=100; end=0;
-            // sub: begin=99; end=-1;
-            // So the kernel will put the slices [99, 0] in reversed order as expected.
-            if (is_reverse && begin <= end) {
-                std::swap(begin, end);
-                begin--;
-                end--;
+        if (params.striding_params.size() == 3) {
+            // If the ith bit of begin_mask is not set, begin[i] is ignored and the range of the appropriate dimension starts from 0.
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/impls/ocl/strided_slice.cpp:140]: Please add some note here this is opposite from the spec, as original comment.

**Review by yeonbok — COMMENTED** [review]
Overall, it looks good to me, if the CI & all the benchmark tests passes.

**Review by yeonbok — APPROVED** [review]

**Comment by yeonbok** [comment]
Please rebase and resolve conflicts.

**Comment by andrew-k-park** [comment]
> Please rebase and resolve conflicts.

Fixed conflicts

**Review by ahnyoung-paul — APPROVED** [review]

**Pull request merged**